appinventor/blocklyeditor/src/backpack.js

```diff
@@ -380,14 +380,11 @@ Blockly.Backpack.prototype.removeFromBackpack = function(ids) {
   var p = this;
   this.getContents(function(/** @type {string[]} */ contents) {
     if (contents && contents.length) {
-      for (var i = 0; i < contents.length; i++) {
-        var xml = Blockly.Xml.textToDom(contents[i]);
-        var blockID = xml.firstElementChild.getAttribute("id");
-        if (ids.indexOf(blockID) >= 0) {
-          contents.splice(i, 1);
-          i--;
-        }
-      }
+      var blockInBackPack = p.workspace_.backpack_.flyout_.workspace_.topBlocks_.map(function(elt) {
```

Comment:
Two things:

1. `p` is already the backpack object, so you can replace `p.workspace_.backpack_.flyout_` with `p.flyout_`
2. Rather than access `topBlocks_` directly, call `getTopBlocks(true)`.